Build failure in externals on macOS M1

On 
```
vavolkl@airm1 cvmfs % sw_vers
ProductName:		macOS
ProductVersion:		13.0
BuildVersion:		22A380
vavolkl@airm1 cvmfs % xcode-select --version
xcode-select version 2396.
vavolkl@airm1 cvmfs % clang --version
Apple clang version 14.0.0 (clang-1400.0.29.202)
Target: arm64-apple-darwin22.1.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
vavolkl@airm1 cvmfs % 
```

there is a build failure in lib archive (see log below). libarchive 3.3.2 builds ok, it seems that it is the patch applied in cvmfs that needs fixing

```
config.status: executing libtool commands
config.status: executing mkdirs commands
/Applications/Xcode.app/Contents/Developer/usr/bin/make  all-am
  CC       libarchive/archive_acl.lo
In file included from libarchive/archive_acl.c:27:
In file included from libarchive/archive_platform.h:84:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/include/inttypes.h:21:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/inttypes.h:229:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/include/stdint.h:52:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdint.h:58:
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/_types/_intmax_t.h:32:25: error: cannot combine with previous 'long long' declaration specifier
typedef __INTMAX_TYPE__ intmax_t;
                        ^
./config.h:1342:23: note: expanded from macro 'intmax_t'
#define intmax_t long long int
                      ^
In file included from libarchive/archive_acl.c:27:
In file included from libarchive/archive_platform.h:84:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/include/inttypes.h:21:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/inttypes.h:229:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/include/stdint.h:52:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdint.h:58:
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/_types/_intmax_t.h:32:25: error: cannot combine with previous 'int' declaration specifier
./config.h:1342:28: note: expanded from macro 'intmax_t'
#define intmax_t long long int
                           ^
In file included from libarchive/archive_acl.c:27:
In file included from libarchive/archive_platform.h:84:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/include/inttypes.h:21:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/inttypes.h:229:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/include/stdint.h:52:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdint.h:59:
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/_types/_uintmax_t.h:32:26: error: cannot combine with previous 'long long' declaration specifier
typedef __UINTMAX_TYPE__ uintmax_t;
                         ^
./config.h:1375:33: note: expanded from macro 'uintmax_t'
#define uintmax_t unsigned long long int
                                ^
In file included from libarchive/archive_acl.c:27:
In file included from libarchive/archive_platform.h:84:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/include/inttypes.h:21:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/inttypes.h:229:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/include/stdint.h:52:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdint.h:59:
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/_types/_uintmax_t.h:32:26: error: cannot combine with previous 'int' declaration specifier
./config.h:1375:38: note: expanded from macro 'uintmax_t'
#define uintmax_t unsigned long long int
                                     ^
In file included from libarchive/archive_acl.c:27:
libarchive/archive_platform.h:112:9: warning: 'SIZE_MAX' macro redefined [-Wmacro-redefined]
#define SIZE_MAX (~(size_t)0)
        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdint.h:173:9: note: previous definition is here
#define SIZE_MAX          UINTPTR_MAX
        ^
In file included from libarchive/archive_acl.c:27:
libarchive/archive_platform.h:118:9: warning: 'UINT32_MAX' macro redefined [-Wmacro-redefined]
#define UINT32_MAX (~(uint32_t)0)
        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdint.h:109:9: note: previous definition is here
#define UINT32_MAX        4294967295U
        ^
In file included from libarchive/archive_acl.c:27:
libarchive/archive_platform.h:121:9: warning: 'INT32_MAX' macro redefined [-Wmacro-redefined]
#define INT32_MAX ((int32_t)(UINT32_MAX >> 1))
        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdint.h:93:9: note: previous definition is here
#define INT32_MAX        2147483647
        ^
In file included from libarchive/archive_acl.c:27:
libarchive/archive_platform.h:124:9: warning: 'INT32_MIN' macro redefined [-Wmacro-redefined]
#define INT32_MIN ((int32_t)(~INT32_MAX))
        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdint.h:104:9: note: previous definition is here
#define INT32_MIN        (-INT32_MAX-1)
        ^
In file included from libarchive/archive_acl.c:27:
libarchive/archive_platform.h:127:9: warning: 'UINT64_MAX' macro redefined [-Wmacro-redefined]
#define UINT64_MAX (~(uint64_t)0)
        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdint.h:110:9: note: previous definition is here
#define UINT64_MAX        18446744073709551615ULL
        ^
In file included from libarchive/archive_acl.c:27:
libarchive/archive_platform.h:130:9: warning: 'INT64_MAX' macro redefined [-Wmacro-redefined]
#define INT64_MAX ((int64_t)(UINT64_MAX >> 1))
        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdint.h:94:9: note: previous definition is here
#define INT64_MAX        9223372036854775807LL
        ^
In file included from libarchive/archive_acl.c:27:
libarchive/archive_platform.h:133:9: warning: 'INT64_MIN' macro redefined [-Wmacro-redefined]
#define INT64_MIN ((int64_t)(~INT64_MAX))
        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdint.h:105:9: note: previous definition is here
#define INT64_MIN        (-INT64_MAX-1)
        ^
In file included from libarchive/archive_acl.c:27:
libarchive/archive_platform.h:136:9: warning: 'UINTMAX_MAX' macro redefined [-Wmacro-redefined]
#define UINTMAX_MAX (~(uintmax_t)0)
        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdint.h:161:9: note: previous definition is here
#define UINTMAX_MAX       UINTMAX_C(18446744073709551615)
        ^
In file included from libarchive/archive_acl.c:27:
libarchive/archive_platform.h:139:9: warning: 'INTMAX_MAX' macro redefined [-Wmacro-redefined]
#define INTMAX_MAX ((intmax_t)(UINTMAX_MAX >> 1))
        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdint.h:160:9: note: previous definition is here
#define INTMAX_MAX        INTMAX_C(9223372036854775807)
        ^
In file included from libarchive/archive_acl.c:27:
libarchive/archive_platform.h:142:9: warning: 'INTMAX_MIN' macro redefined [-Wmacro-redefined]
#define INTMAX_MIN ((intmax_t)(~INTMAX_MAX))
        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdint.h:162:9: note: previous definition is here
#define INTMAX_MIN        (-INTMAX_MAX-1)
        ^
In file included from libarchive/archive_acl.c:37:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/wchar.h:75:
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/_types/_size_t.h:31:32: error: cannot combine with previous 'type-name' declaration specifier
typedef __darwin_size_t        size_t;
                               ^
./config.h:1351:25: note: expanded from macro 'size_t'
#define size_t unsigned int
                        ^
In file included from libarchive/archive_acl.c:37:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/wchar.h:75:
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/_types/_size_t.h:31:32: error: 'type-name' cannot be signed or unsigned
./config.h:1351:16: note: expanded from macro 'size_t'
#define size_t unsigned int
               ^
In file included from libarchive/archive_acl.c:37:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/wchar.h:90:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdio.h:334:
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/_types/_off_t.h:31:33: error: 'long long type-name' is invalid
typedef __darwin_off_t          off_t;
                                ^
./config.h:1348:15: note: expanded from macro 'off_t'
#define off_t long long
              ^
In file included from libarchive/archive_acl.c:40:
In file included from libarchive/archive_acl_private.h:35:
In file included from libarchive/archive_string.h:40:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdlib.h:66:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/wait.h:90:
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/_types/_id_t.h:31:25: error: 'type-name' cannot be signed or unsigned
typedef __darwin_id_t   id_t;           /* can hold pid_t, gid_t, or uid_t */
                        ^
./config.h:1326:14: note: expanded from macro 'id_t'
#define id_t unsigned long
             ^
In file included from libarchive/archive_acl.c:40:
In file included from libarchive/archive_acl_private.h:35:
In file included from libarchive/archive_string.h:40:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdlib.h:66:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/wait.h:90:
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/_types/_id_t.h:31:25: error: 'long type-name' is invalid
./config.h:1326:23: note: expanded from macro 'id_t'
#define id_t unsigned long
                      ^
In file included from libarchive/archive_acl.c:40:
In file included from libarchive/archive_acl_private.h:35:
In file included from libarchive/archive_string.h:40:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdlib.h:66:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/wait.h:109:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/signal.h:156:
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/_types/_uid_t.h:31:31: error: cannot combine with previous 'type-name' declaration specifier
typedef __darwin_uid_t        uid_t;
                              ^
./config.h:1355:15: note: expanded from macro 'uid_t'
#define uid_t int
              ^
In file included from libarchive/archive_acl.c:40:
In file included from libarchive/archive_acl_private.h:35:
In file included from libarchive/archive_string.h:40:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdlib.h:257:
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/_types/_mode_t.h:31:33: error: cannot combine with previous 'type-name' declaration specifier
typedef __darwin_mode_t         mode_t;
                                ^
./config.h:1345:16: note: expanded from macro 'mode_t'
#define mode_t int
               ^
In file included from libarchive/archive_acl.c:40:
In file included from libarchive/archive_acl_private.h:35:
In file included from libarchive/archive_string.h:49:
In file included from libarchive/archive.h:41:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/stat.h:97:
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/_types/_gid_t.h:31:25: error: cannot combine with previous 'type-name' declaration specifier
typedef __darwin_gid_t  gid_t;
                        ^
./config.h:1323:15: note: expanded from macro 'gid_t'
#define gid_t int
              ^
10 warnings and 12 errors generated.
make[1]: *** [libarchive/archive_acl.lo] Error 1
make: *** [all] Error 2
/Applications/Xcode.app/Contents/Developer/usr/bin/make  install-am
  CC       libarchive/archive_acl.lo
In file included from libarchive/archive_acl.c:27:
In file included from libarchive/archive_platform.h:84:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/include/inttypes.h:21:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/inttypes.h:229:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/include/stdint.h:52:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdint.h:58:
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/_types/_intmax_t.h:32:25: error: cannot combine with previous 'long long' declaration specifier
typedef __INTMAX_TYPE__ intmax_t;
                        ^
./config.h:1342:23: note: expanded from macro 'intmax_t'
#define intmax_t long long int
                      ^
In file included from libarchive/archive_acl.c:27:
In file included from libarchive/archive_platform.h:84:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/include/inttypes.h:21:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/inttypes.h:229:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/include/stdint.h:52:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdint.h:58:
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/_types/_intmax_t.h:32:25: error: cannot combine with previous 'int' declaration specifier
./config.h:1342:28: note: expanded from macro 'intmax_t'
#define intmax_t long long int
                           ^
In file included from libarchive/archive_acl.c:27:
In file included from libarchive/archive_platform.h:84:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/include/inttypes.h:21:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/inttypes.h:229:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/include/stdint.h:52:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdint.h:59:
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/_types/_uintmax_t.h:32:26: error: cannot combine with previous 'long long' declaration specifier
typedef __UINTMAX_TYPE__ uintmax_t;
                         ^
./config.h:1375:33: note: expanded from macro 'uintmax_t'
#define uintmax_t unsigned long long int
                                ^
In file included from libarchive/archive_acl.c:27:
In file included from libarchive/archive_platform.h:84:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/include/inttypes.h:21:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/inttypes.h:229:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/include/stdint.h:52:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdint.h:59:
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/_types/_uintmax_t.h:32:26: error: cannot combine with previous 'int' declaration specifier
./config.h:1375:38: note: expanded from macro 'uintmax_t'
#define uintmax_t unsigned long long int
                                     ^
In file included from libarchive/archive_acl.c:27:
libarchive/archive_platform.h:112:9: warning: 'SIZE_MAX' macro redefined [-Wmacro-redefined]
#define SIZE_MAX (~(size_t)0)
        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdint.h:173:9: note: previous definition is here
#define SIZE_MAX          UINTPTR_MAX
        ^
In file included from libarchive/archive_acl.c:27:
libarchive/archive_platform.h:118:9: warning: 'UINT32_MAX' macro redefined [-Wmacro-redefined]
#define UINT32_MAX (~(uint32_t)0)
        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdint.h:109:9: note: previous definition is here
#define UINT32_MAX        4294967295U
        ^
In file included from libarchive/archive_acl.c:27:
libarchive/archive_platform.h:121:9: warning: 'INT32_MAX' macro redefined [-Wmacro-redefined]
#define INT32_MAX ((int32_t)(UINT32_MAX >> 1))
        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdint.h:93:9: note: previous definition is here
#define INT32_MAX        2147483647
        ^
In file included from libarchive/archive_acl.c:27:
libarchive/archive_platform.h:124:9: warning: 'INT32_MIN' macro redefined [-Wmacro-redefined]
#define INT32_MIN ((int32_t)(~INT32_MAX))
        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdint.h:104:9: note: previous definition is here
#define INT32_MIN        (-INT32_MAX-1)
        ^
In file included from libarchive/archive_acl.c:27:
libarchive/archive_platform.h:127:9: warning: 'UINT64_MAX' macro redefined [-Wmacro-redefined]
#define UINT64_MAX (~(uint64_t)0)
        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdint.h:110:9: note: previous definition is here
#define UINT64_MAX        18446744073709551615ULL
        ^
In file included from libarchive/archive_acl.c:27:
libarchive/archive_platform.h:130:9: warning: 'INT64_MAX' macro redefined [-Wmacro-redefined]
#define INT64_MAX ((int64_t)(UINT64_MAX >> 1))
        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdint.h:94:9: note: previous definition is here
#define INT64_MAX        9223372036854775807LL
        ^
In file included from libarchive/archive_acl.c:27:
libarchive/archive_platform.h:133:9: warning: 'INT64_MIN' macro redefined [-Wmacro-redefined]
#define INT64_MIN ((int64_t)(~INT64_MAX))
        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdint.h:105:9: note: previous definition is here
#define INT64_MIN        (-INT64_MAX-1)
        ^
In file included from libarchive/archive_acl.c:27:
libarchive/archive_platform.h:136:9: warning: 'UINTMAX_MAX' macro redefined [-Wmacro-redefined]
#define UINTMAX_MAX (~(uintmax_t)0)
        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdint.h:161:9: note: previous definition is here
#define UINTMAX_MAX       UINTMAX_C(18446744073709551615)
        ^
In file included from libarchive/archive_acl.c:27:
libarchive/archive_platform.h:139:9: warning: 'INTMAX_MAX' macro redefined [-Wmacro-redefined]
#define INTMAX_MAX ((intmax_t)(UINTMAX_MAX >> 1))
        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdint.h:160:9: note: previous definition is here
#define INTMAX_MAX        INTMAX_C(9223372036854775807)
        ^
In file included from libarchive/archive_acl.c:27:
libarchive/archive_platform.h:142:9: warning: 'INTMAX_MIN' macro redefined [-Wmacro-redefined]
#define INTMAX_MIN ((intmax_t)(~INTMAX_MAX))
        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdint.h:162:9: note: previous definition is here
#define INTMAX_MIN        (-INTMAX_MAX-1)
        ^
In file included from libarchive/archive_acl.c:37:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/wchar.h:75:
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/_types/_size_t.h:31:32: error: cannot combine with previous 'type-name' declaration specifier
typedef __darwin_size_t        size_t;
                               ^
./config.h:1351:25: note: expanded from macro 'size_t'
#define size_t unsigned int
                        ^
In file included from libarchive/archive_acl.c:37:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/wchar.h:75:
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/_types/_size_t.h:31:32: error: 'type-name' cannot be signed or unsigned
./config.h:1351:16: note: expanded from macro 'size_t'
#define size_t unsigned int
               ^
In file included from libarchive/archive_acl.c:37:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/wchar.h:90:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdio.h:334:
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/_types/_off_t.h:31:33: error: 'long long type-name' is invalid
typedef __darwin_off_t          off_t;
                                ^
./config.h:1348:15: note: expanded from macro 'off_t'
#define off_t long long
              ^
In file included from libarchive/archive_acl.c:40:
In file included from libarchive/archive_acl_private.h:35:
In file included from libarchive/archive_string.h:40:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdlib.h:66:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/wait.h:90:
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/_types/_id_t.h:31:25: error: 'type-name' cannot be signed or unsigned
typedef __darwin_id_t   id_t;           /* can hold pid_t, gid_t, or uid_t */
                        ^
./config.h:1326:14: note: expanded from macro 'id_t'
#define id_t unsigned long
             ^
In file included from libarchive/archive_acl.c:40:
In file included from libarchive/archive_acl_private.h:35:
In file included from libarchive/archive_string.h:40:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdlib.h:66:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/wait.h:90:
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/_types/_id_t.h:31:25: error: 'long type-name' is invalid
./config.h:1326:23: note: expanded from macro 'id_t'
#define id_t unsigned long
                      ^
In file included from libarchive/archive_acl.c:40:
In file included from libarchive/archive_acl_private.h:35:
In file included from libarchive/archive_string.h:40:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdlib.h:66:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/wait.h:109:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/signal.h:156:
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/_types/_uid_t.h:31:31: error: cannot combine with previous 'type-name' declaration specifier
typedef __darwin_uid_t        uid_t;
                              ^
./config.h:1355:15: note: expanded from macro 'uid_t'
#define uid_t int
              ^
In file included from libarchive/archive_acl.c:40:
In file included from libarchive/archive_acl_private.h:35:
In file included from libarchive/archive_string.h:40:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdlib.h:257:
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/_types/_mode_t.h:31:33: error: cannot combine with previous 'type-name' declaration specifier
typedef __darwin_mode_t         mode_t;
                                ^
./config.h:1345:16: note: expanded from macro 'mode_t'
#define mode_t int
               ^
In file included from libarchive/archive_acl.c:40:
In file included from libarchive/archive_acl_private.h:35:
In file included from libarchive/archive_string.h:49:
In file included from libarchive/archive.h:41:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/stat.h:97:
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/_types/_gid_t.h:31:25: error: cannot combine with previous 'type-name' declaration specifier
typedef __darwin_gid_t  gid_t;
                        ^
./config.h:1323:15: note: expanded from macro 'gid_t'
#define gid_t int
              ^
10 warnings and 12 errors generated.
make[1]: *** [libarchive/archive_acl.lo] Error 1
make: *** [install] Error 2
CMake Error at CMakeLists.txt:208 (message):
  bootstrapping failed

```

--- a/externals/libarchive/src/fix-new-glibc.patch
+++ b/externals/libarchive/src/fix-new-glibc.patch
@@ -14,7 +14,7 @@
  #endif
  #ifdef HAVE_LINUX_FS_H
  #include <linux/fs.h>
-+#elif HAVE_SYS_MOUNT_H
++#elifdef HAVE_SYS_MOUNT_H
 +#include <sys/mount.h>
  #endif
  /*

--- a/externals/libarchive/src/fix-new-glibc.patch
+++ b/externals/libarchive/src/fix-new-glibc.patch
@@ -14,7 +14,7 @@
  #endif
  #ifdef HAVE_LINUX_FS_H
  #include <linux/fs.h>
-+#elifdef HAVE_SYS_MOUNT_H
++#elif defined HAVE_SYS_MOUNT_H
 +#include <sys/mount.h>
  #endif
  /*

